pkg/e2e/infra/infra.go

```diff
@@ -112,119 +111,88 @@ var _ = g.Describe("[Feature:Machines] Machines should", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 	})
 
-	g.It("be replaced with new one when deleted", func() {
+	g.It("recover from deleted worker machines", func() {
 		var err error
 		client, err := e2e.LoadClient()
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		// Assume cluster state
+		// Expect for cluster to cool down from previous tests
 		err = e2e.WaitUntilAllNodesAreReady(client)
```

Comment:
This is still no assurance. Some nodes may turn to NotReady state after this line if the corresponding machine is deleted in a previous test. As long we we don't remove a node right before a machine is deleted, this will never be 100%.